src/staticanalysis/bot/static_analysis_bot/revisions.py

```diff
@@ -134,33 +134,74 @@ def url(self):
 
     def load(self, repo):
         '''
-        Load patch from Phabricator
+        Load full stack of patches from Phabricator:
+        * setup repo to base revision from Mozilla Central
+        * Apply previous needed patches from Phabricator
         '''
         assert isinstance(repo, hglib.client.hgclient)
 
-        # Load raw patch
-        self.patch = self.api.load_raw_diff(self.diff_id)
+        # Diff PHIDs from our patch to its base
+        patches = OrderedDict()
+        patches[self.diff_phid] = self.diff_id
 
-    def apply(self, repo):
-        '''
-        Apply patch from Phabricator to Mercurial local repository
-        '''
-        assert isinstance(repo, hglib.client.hgclient)
+        parents = self.api.load_parents(self.phid)
+        if parents:
+
+            # Load all parent diffs
+            for parent in parents:
+                logger.info('Loading parent diff', phid=parent)
+                parent_diffs = self.api.search_diffs(revision_phid=parent)
+
+                # Use only the first (last in time) diff for each revision
```

Comment:
Two questions:
1) Maybe we should get the parent of the specific diff? There could be cases where the analysis is triggered, then the developer uploads a new stack of revisions and we get wrong results. I'm sure this is really rare, but in theory it could happen.
2) If we don't do (1), we should file an issue about it so we don't forget. Then, are we sure the first is always the last in time? Maybe we should sort by ID so that we're sure we're getting the latest?

Reply:
1 is not possible, `edge.search` only gives the revisions
I'm doing 2